### PR TITLE
Add handwritten sweeper for  google_vmwareengine_private_cloud

### PR DESCRIPTION
**Pull request by SarahFrench**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/18485

This PR adds a sweeper for private cloud resources. Note that PCs can have long deletion times - to speed this up[ I pass parameters into the DELETE HTTP request to try speed up deletion.](https://github.com/GoogleCloudPlatform/magic-modules/pull/11002#discussion_r1647648806)

Here are some builds in TeamCity using this sweeper: https://hashicorp.teamcity.com/buildConfiguration/TerraformProviders_GoogleCloud_GOOGLE_MMUPSTREAMTESTS_SERVICE_SWEEPER__MANUAL?branch=refs%2Fheads%2Fauto-pr-11002&buildTypeTab=overview&mode=builds

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: mmv1/third_party/terraform/services/vmwareengine/resource_vmwareengine_private_cloud_sweeper.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+	}
+
+	t := &testing.T{}
+	billingId := envvar.GetTestBillingAccountFromEnv(t)
+
+	// List of location values includes:
+	//   * global location
+	//   * regions used for this resource type's acc tests in the past
+	//   * the 'region' passed to the sweeper
+	locations := []string{"global", "us-central1", "southamerica-west1", "me-west1", region}
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/services/vmwareengine/resource_vmwareengine_private_cloud_sweeper.go:44]:
```suggestion
	locations := []string{region, "global", "southamerica-west1", "me-west1"}
```

Use user-provided region first, remove us-central1 as it's assumed that's the value of region most times

File: mmv1/third_party/terraform/services/vmwareengine/resource_vmwareengine_network_sweeper.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+	}
+
+	t := &testing.T{}
+	billingId := envvar.GetTestBillingAccountFromEnv(t)
+
+	// List of location values includes:
+	//   * global location
+	//   * regions used for this resource type's acc tests in the past
+	//   * the 'region' passed to the sweeper
+	locations := []string{"global", "us-central1", "southamerica-west1", "me-west1", region}
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/services/vmwareengine/resource_vmwareengine_network_sweeper.go:43]:
```suggestion
	locations := []string{region, "global", "southamerica-west1", "me-west1"}
```

Use user-provided region first, remove us-central1 as it's assumed that's the value of region most times

File: mmv1/third_party/terraform/services/vmwareengine/resource_vmwareengine_network_sweeper.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+	for _, location := range locations {
+		log.Printf("[INFO][SWEEPER_LOG] Beginning the process of sweeping location '%s'.", location)
+
+		// Setup variables to replace in list template
+		d := &tpgresource.ResourceDataMock{
+			FieldsInSchema: map[string]interface{}{
+				"project":         config.Project,
+				"region":          location,
+				"location":        location,
+				"zone":            "-",
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/services/vmwareengine/resource_vmwareengine_network_sweeper.go:53]:
```suggestion
				"zone":            location,
```

File: mmv1/third_party/terraform/services/vmwareengine/resource_vmwareengine_private_cloud_sweeper.go [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+	t := &testing.T{}
+	billingId := envvar.GetTestBillingAccountFromEnv(t)
+
+	// List of location values includes:
+	//   * global location
+	//   * regions used for this resource type's acc tests in the past
+	//   * the 'region' passed to the sweeper
+	locations := []string{region, "southamerica-west1-a", "me-west1-a"}
+	for _, location := range locations {
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/services/vmwareengine/resource_vmwareengine_private_cloud_sweeper.go:45]:
```suggestion
	locations := []string{region, "southamerica-west1-a", "me-west1-a"}
	log.Printf("[INFO][SWEEPER_LOG] Sweeping will include these locations: %v.", locations)
	for _, location := range locations {
```

File: mmv1/third_party/terraform/services/vmwareengine/resource_vmwareengine_private_cloud_sweeper.go [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+	if err != nil {
+		log.Printf("[INFO][SWEEPER_LOG] error loading: %s", err)
+		return err
+	}
+
+	t := &testing.T{}
+	billingId := envvar.GetTestBillingAccountFromEnv(t)
+
+	// List of location values includes:
+	//   * global location
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/services/vmwareengine/resource_vmwareengine_private_cloud_sweeper.go:41]:
```suggestion
```

File: mmv1/third_party/terraform/services/vmwareengine/resource_vmwareengine_private_cloud_sweeper.go [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+	//   * regions used for this resource type's acc tests in the past
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/services/vmwareengine/resource_vmwareengine_private_cloud_sweeper.go:42]:
```suggestion
	//   * zones used for this resource type's acc tests in the past
```

File: mmv1/third_party/terraform/services/vmwareengine/resource_vmwareengine_private_cloud_sweeper.go [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+			if err != nil {
+				log.Printf("[INFO][SWEEPER_LOG] error preparing delete url: %s", err)
+				continue
+			}
+			deleteUrl = deleteUrl + name
+
+			// We force delete the Private Cloud and ensure there's no delay in deletion
+			force := true
+			delayHours := 0
+			deleteUrl = deleteUrl + fmt.Sprintf("?force=%t&delayHours=%d", force, delayHours)
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/services/vmwareengine/resource_vmwareengine_private_cloud_sweeper.go:115]: Please see this section of the docs: https://cloud.google.com/vmware-engine/docs/reference/rest/v1/projects.locations.privateClouds/delete#query-parameters

These params force deletion of child resources (e.g. clusters) and make deletion faster:

> Specifying a value of 0 for this field instead begins the deletion process and ceases billing immediately. During the final deletion process, the value of PrivateCloud.state becomes PURGING.

**Review by zli82016 — APPROVED** [review]
LGTM. Thanks for adding the sweeper.

**Comment by SarahFrench** [comment]
Thanks! I'll just rerun the checks as they appear to have got stuck

/gcbrun

**Pull request merged**